guides/source/security.md

```diff
@@ -29,6 +30,147 @@ The threats against web applications include user account hijacking, bypass of a
 
 In order to develop secure web applications you have to keep up to date on all layers and know your enemies. To keep up to date subscribe to security mailing lists, read security blogs, and make updating and security checks a habit (check the [Additional Resources](#additional-resources) chapter). It is done manually because that's how you find the nasty logical security problems.
 
+Authentication
+--------------
+
+Authentication is often one the first features implemented in a web application. It serves as the foundation for securing user data and is part of most modern web applications.
+
+Starting with version 8.0, Rails comes with a default authentication generator, which provides a solid starting point for securing your application by only allowing access to verified users.
+
+The authentication generator adds all of the relevant models, controllers, views, routes, and migrations needed for basic authentication and password reset functionality.
+
+To use this feature in your application, you can run `rails generate authentication`. Here are all of the files the generator modifies and new files it adds:
+
+```bash
+$ rails generate authentication
+      invoke  erb
+      create    app/views/passwords/new.html.erb
+      create    app/views/passwords/edit.html.erb
+      create    app/views/sessions/new.html.erb
+      create  app/models/session.rb
+      create  app/models/user.rb
+      create  app/models/current.rb
+      create  app/controllers/sessions_controller.rb
+      create  app/controllers/concerns/authentication.rb
+      create  app/controllers/passwords_controller.rb
+      create  app/mailers/passwords_mailer.rb
+      create  app/views/passwords_mailer/reset.html.erb
+      create  app/views/passwords_mailer/reset.text.erb
+      create  test/mailers/previews/passwords_mailer_preview.rb
+        gsub  app/controllers/application_controller.rb
+       route  resources :passwords, param: :token
+       route  resource :session
+        gsub  Gemfile
+      bundle  install --quiet
+    generate  migration CreateUsers email_address:string!:uniq password_digest:string! --force
+       rails  generate migration CreateUsers email_address:string!:uniq password_digest:string! --force
+      invoke  active_record
+      create    db/migrate/20241010215312_create_users.rb
+    generate  migration CreateSessions user:references ip_address:string user_agent:string --force
+       rails  generate migration CreateSessions user:references ip_address:string user_agent:string --force
+      invoke  active_record
+      create    db/migrate/20241010215314_create_sessions.rb
+```
+
+As shown above, the authentication generator modifies the `Gemfile` to add the `bcrypt` gem. The generator uses the `bcrypt` gem for storing a hash of the password in the database (instead of plain-text passwords). As this process is not reversible, there's no way to go from the hash back to the password.
```

Comment:
```suggestion
As shown above, the authentication generator modifies the `Gemfile` to add the [bcrypt](https://github.com/bcrypt-ruby/bcrypt-ruby/) gem. The generator uses the `bcrypt` gem for storing a hash of the password in the database (instead of plain-text passwords). As this process is not reversible, there's no way to go from the hash back to the password.
```
I still believe it's good to have, but it would be better to redirect the user directly to the source link.